### PR TITLE
Add MarkdownInfo entry in Module Concrete Decl and proper errors

### PR DESCRIPTION
**Pull request by jonaprieto**
Remove TODO added by @janmasrovira in 
- https://github.com/anoma/juvix/pull/2513

### CODE REVIEW

File: src/Juvix/Compiler/Backend/Markdown/Data/Types.hs [hunk 1]
```diff
@@ -234,6 +234,11 @@ instance-- (MK.IsInline TextBlock) =>
           xs
       )
 
+nullMk :: Mk -> Bool
+nullMk = \case
+  MkNull -> True
+  _ -> False
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Backend/Markdown/Data/Types.hs:240]: what about concat null null?

File: src/Juvix/Compiler/Backend/Markdown/Translation/FromTyped/Source.hs [hunk 2]
```diff
@@ -55,8 +57,22 @@ fromJuvixMarkdown opts = do
       m :: Concrete.Module 'Concrete.Scoped 'Concrete.ModuleTop
       m = opts ^. processJuvixBlocksArgsModule
 
-  case (m ^. Concrete.moduleMarkdown, m ^. Concrete.moduleMarkdownSeparation) of
-    (Just mk, Just sepr) -> do
+      fname :: Path Abs File
+      fname = getLoc m ^. intervalFile
+
+  case (m ^. Concrete.moduleMarkdownInfo) of
+    (Just mkInfo) -> do
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Backend/Markdown/Translation/FromTyped/Source.hs:64]:
```suggestion
  case m ^. Concrete.moduleMarkdownInfo of
    Just mkInfo -> do
```

File: src/Juvix/Compiler/Concrete/Translation/FromSource.hs [hunk 3]
```diff
@@ -141,7 +142,10 @@ runModuleParser fileName input
       res <- P.runParserT juvixCodeBlockParser (toFilePath fileName) input
       case res of
         Left err -> return . Left . ErrMegaparsec . MegaparsecError $ err
-        Right r -> runMarkdownModuleParser fileName r
+        Right r -> do
+          if
+              | MK.nullMk r -> return . Left . ErrMarkdownBackend $ ErrNoJuvixCodeBlocks NoJuvixCodeBlocksError {_noJuvixCodeBlocksErrorFilepath = fileName}
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Concrete/Translation/FromSource.hs:147]:
```suggestion
        Right
              | MK.nullMk r -> return . Left . ErrMarkdownBackend $ ErrNoJuvixCodeBlocks NoJuvixCodeBlocksError {_noJuvixCodeBlocksErrorFilepath = fileName}
```

Comment (janmasrovira) [inline src/Juvix/Compiler/Concrete/Translation/FromSource.hs:147]:
```suggestion
        Right
              | MK.nullMk r -> return . Left . ErrMarkdownBackend $ ErrNoJuvixCodeBlocks NoJuvixCodeBlocksError {_noJuvixCodeBlocksErrorFilepath = fileName}
```

File: src/Juvix/Compiler/Backend/Markdown/Error.hs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+module Juvix.Compiler.Backend.Markdown.Error where
+
+import Juvix.Compiler.Concrete.Language
+import Juvix.Compiler.Concrete.Translation.FromParsed.Analysis.Scoping.Error.Pretty
+import Juvix.Prelude
+
+data MarkdownBackendError
+  = ErrNoMarkdownInfo NoMarkdownInfoError
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Backend/Markdown/Error.hs:8]: It needs to be clear that this is an internal error

File: app/Commands/Markdown.hs [hunk 5]
```diff
@@ -20,35 +20,38 @@ runCommand opts = do
...
+                opts ^. markdownNoPath,
+              _processJuvixBlocksArgsComments = scopedM ^. Scoper.comments,
+              _processJuvixBlocksArgsModule = m,
+              _processJuvixBlocksArgsOutputDir = outputDir
+            }
+  case res of
+    Left err -> exitJuvixError (JuvixError err)
+    Right md ->
+      if
+          | opts ^. markdownStdout -> liftIO . putStrLn $ md
```
Comment (janmasrovira) [inline app/Commands/Markdown.hs:40]:
```suggestion
    Right md
          | opts ^. markdownStdout -> liftIO . putStrLn $ md
```

File: src/Juvix/Compiler/Backend/Markdown/Data/Types.hs (hunk 1 above)
Comment (jonaprieto) [reply src/Juvix/Compiler/Backend/Markdown/Data/Types.hs:240]: I was using it internally after the mk is reduced, but we can it more general. np.

Comment (jonaprieto) [reply src/Juvix/Compiler/Backend/Markdown/Data/Types.hs]: Done

File: src/Juvix/Compiler/Backend/Markdown/Translation/FromTyped/Source.hs (hunk 2 above)
Comment (jonaprieto) [reply src/Juvix/Compiler/Backend/Markdown/Translation/FromTyped/Source.hs]: Done

File: src/Juvix/Compiler/Concrete/Translation/FromSource.hs (hunk 3 above)
Comment (jonaprieto) [reply src/Juvix/Compiler/Concrete/Translation/FromSource.hs]: Done

Comment (jonaprieto) [reply src/Juvix/Compiler/Concrete/Translation/FromSource.hs]: Done

File: src/Juvix/Compiler/Backend/Markdown/Error.hs (hunk 4 above)
Comment (jonaprieto) [reply src/Juvix/Compiler/Backend/Markdown/Error.hs]: Done

File: app/Commands/Markdown.hs (hunk 5 above)
Comment (jonaprieto) [reply app/Commands/Markdown.hs]: Done

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**